staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_interval.go

```diff
@@ -133,9 +134,22 @@ func (s sortableWatchCacheEvents) Swap(i, j int) {
 // returned by Next() need to be events from a List() done on the underlying store of
 // the watch cache.
 // The items returned in the interval will be sorted by Key.
-func newCacheIntervalFromStore(resourceVersion uint64, store cache.Indexer, getAttrsFunc attrFunc) (*watchCacheInterval, error) {
+func newCacheIntervalFromStore(resourceVersion uint64, store cache.Indexer, getAttrsFunc attrFunc, key string, predicate *storage.SelectionPredicate) (*watchCacheInterval, error) {
 	buffer := &watchCacheIntervalBuffer{}
-	allItems := store.List()
+	var allItems []interface{}
+
+	if _, ok := predicate.MatchesSingle(); ok {
```

Comment:
what if predicate is nil?

In general, I'm not a fan of passing predicate here - it's tempting to use other stuff here. Can we instead pass "matchesSingle" bool value instead?

Reply:
Given the code paths that call `newCacheIntervalFromStore`, `predicate`  is always non-nil as a struct element of another non-nil struct. But I agree making it simple is preferable. Changed and thanks for the suggestion.